bokehjs/src/lib/models/layouts/layout_dom.ts

```diff
@@ -156,25 +156,20 @@ export abstract class LayoutDOMView extends PaneView {
     const created = await this.build_child_views()
     const created_children = new Set(created)
 
-    // First remove and then either reattach existing elements or render and
-    // attach new elements, so that the order of children is consistent, while
-    // avoiding expensive re-rendering of existing views.
-    for (const child_view of this.child_views) {
-      child_view.el.remove()
-    }
-
+    // Since appending to a DOM node will move the node to the end if it has
+    // already been added appending all the children in order will result in
+    // correct ordering
     for (const child_view of this.child_views) {
       const is_new = created_children.has(child_view)
-
       const target = child_view.rendering_target() ?? this.shadow_el
       if (is_new) {
         child_view.render_to(target)
       } else {
-        target.append(child_view.el)
+        target.appendChild(child_view.el)
```

Comment:
`.append()` doesn't work as expected or this change a leftover from your previous changes?

Reply:
It works fine, I can revert.